src/QsFmt/Formatter/ParseTree/Statement.fs

```diff
@@ -51,6 +99,62 @@ type StatementVisitor(tokens) =
         }
         |> Return
 
+    override _.VisitUseStatement context =
+        {
+            Kind = Use
+            Keyword = context.``use`` |> Node.toTerminal tokens
+            Binding = context.binding |> qubitBindingVisitor.Visit
+            OpenParen = context.openParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            CloseParen = context.closeParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            Coda = context.semicolon |> Node.toTerminal tokens |> Semicolon
+        }
+        |> QubitDeclaration
+
+    override visitor.VisitUseBlock context =
+        {
+            Kind = Use
+            Keyword = context.``use`` |> Node.toTerminal tokens
+            Binding = context.binding |> qubitBindingVisitor.Visit
+            OpenParen = context.openParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            CloseParen = context.closeParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            Coda =
+                {
+                    OpenBrace = context.body.openBrace |> Node.toTerminal tokens
+                    Items = context.body._statements |> Seq.map visitor.Visit |> List.ofSeq
+                    CloseBrace = context.body.closeBrace |> Node.toTerminal tokens
+                }
+                |> Block
+        }
+        |> QubitDeclaration
+
+    override _.VisitBorrowStatement context =
+        {
+            Kind = Borrow
+            Keyword = context.borrow |> Node.toTerminal tokens
+            Binding = context.binding |> qubitBindingVisitor.Visit
+            OpenParen = context.openParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            CloseParen = context.closeParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            Coda = context.semicolon |> Node.toTerminal tokens |> Semicolon
+        }
+        |> QubitDeclaration
+
+    override visitor.VisitBorrowBlock context =
+        {
+            Kind = Borrow
+            Keyword = context.borrow |> Node.toTerminal tokens
+            Binding = context.binding |> qubitBindingVisitor.Visit
+            OpenParen = context.openParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            CloseParen = context.closeParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
+            Coda =
+                {
+                    OpenBrace = context.body.openBrace |> Node.toTerminal tokens
+                    Items = context.body._statements |> Seq.map visitor.Visit |> List.ofSeq
+                    CloseBrace = context.body.closeBrace |> Node.toTerminal tokens
+                }
+                |> Block
+        }
+        |> QubitDeclaration
```

Comment:
nit: You could factor these out so there's less duplicate code (either by changing the grammar like I mentioned in the previous comment, or by writing a couple helper functions here to handle the things in common).

Reply:
These were merged into one handle for the one rule.